compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmCodeGenUtil.java

```diff
@@ -507,30 +511,70 @@ public static boolean isBallerinaBuiltinModule(String orgName, String moduleName
         return orgName.equals("ballerina") && moduleName.equals("builtin");
     }
 
-    public static void generateBbInstructions(MethodVisitor mv, LabelGenerator labelGen, JvmInstructionGen instGen,
-                                              int localVarOffset, AsyncDataCollector asyncDataCollector,
-                                              String funcName,
-                                              BIRNode.BIRBasicBlock bb) {
+    public static BirScope generateBbInstructions(MethodVisitor mv, LabelGenerator labelGen, JvmInstructionGen instGen,
+                                                  int localVarOffset, AsyncDataCollector asyncDataCollector,
+                                                  String funcName,
+                                                  BIRNode.BIRBasicBlock bb, Set<BirScope> visitedScopesSet,
+                                                  BirScope lastScope) {
+
         int insCount = bb.instructions.size();
         for (int i = 0; i < insCount; i++) {
             Label insLabel = labelGen.getLabel(funcName + bb.id.value + "ins" + i);
             mv.visitLabel(insLabel);
             BIRInstruction inst = bb.instructions.get(i);
             if (inst != null) {
-                generateDiagnosticPos(((BIRNode) inst).pos, mv);
+                lastScope = generateDiagnosticPos((BIRAbstractInstruction) inst, funcName, mv, labelGen,
+                                                  visitedScopesSet, lastScope);
                 instGen.generateInstructions(localVarOffset, asyncDataCollector, inst);
             }
         }
+
+        return lastScope;
+    }
+
+    private static BirScope generateDiagnosticPos(BIRAbstractInstruction instruction, String funcName, MethodVisitor mv,
```

Comment:
Same here as well. Method name may not be correct as it returns BirScope?

Reply:
Fixed